### PR TITLE
[manager ]feat(manager): mutate field level state on validation error

### PR DESCRIPTION
**Pull request by Hyperkid123**
part of #687

### CODE REVIEW

File: packages/form-state-manager/src/utils/use-subscription.ts [hunk 1]
```diff
@@ -74,12 +74,37 @@ const useSubscription = ({ name, initialValue, clearOnUnmount, initializeOnMount
...
+        validating: false
+      }
+    }));
+  };
+
   const handleChange = (event: OnChangeEvent) => {
     const sanitizedValue = sanitizeValue(event);
     change(name, sanitizedValue);
-    // TODO Mutate field state with validation flags
+    // TODO Memoize validation results
```
Comment (rvsia) [inline packages/form-state-manager/src/utils/use-subscription.ts:93]: could be opt-in.. I can imagine that sometimes you want to check the same value (async validation in very dynamic applications)

**Comment by Hyperkid123** [comment]
> Code looks good... but I have one request:

Could you please add a test for multiple async validations at the same time to be sure it works correctly? (so its being validated until the last validation is resolved?)

@rvsia its not ready for that. It will need a similar construct as in the manager API. I plan to do that later on after the global manager API changes as well.

File: packages/form-state-manager/src/utils/use-subscription.ts (hunk 1 above)
Comment (Hyperkid123) [reply packages/form-state-manager/src/utils/use-subscription.ts:93]: Yes. But I would rather make it opt-out.

**Pull request merged**